translation-web/src/main/java/eu/europeana/api/translation/web/exception/GlobalExceptionHandler.java

```diff
@@ -1,19 +1,82 @@
 import eu.europeana.api.commons.config.i18n.I18nService;
-import eu.europeana.api.commons.web.controller.exception.AbstractExceptionHandlingController;
+import eu.europeana.api.commons.error.EuropeanaApiErrorResponse;
+import eu.europeana.api.commons.web.exception.EuropeanaGlobalExceptionHandler;
+import eu.europeana.api.commons.web.exception.HttpException;
 import eu.europeana.api.translation.config.BeanNames;
+import eu.europeana.api.translation.definitions.vocabulary.TranslationAppConstants;
+import eu.europeana.api.translation.web.service.RequestPathMethodService;
 
 @ControllerAdvice
-public class GlobalExceptionHandler extends AbstractExceptionHandlingController {
-
-	@Resource(name=BeanNames.BEAN_I18N_SERVICE)
+@ConditionalOnWebApplication
+public class GlobalExceptionHandler extends EuropeanaGlobalExceptionHandler {
+  
 	I18nService i18nService;
 
 	protected I18nService getI18nService() {
 		return i18nService;
 	}
+	
+	@Autowired
+	public GlobalExceptionHandler(
+	    RequestPathMethodService requestPathMethodService, 
+	    @Qualifier(BeanNames.BEAN_I18N_SERVICE) I18nService i18nService) {
+	  this.requestPathMethodService = requestPathMethodService;
+	  this.i18nService = i18nService;
+	}
+
+	  /**
+	   * Default handler for EuropeanaApiException types
+	   *
+	   * @param e caught exception
+	   */
+	  @ExceptionHandler
+	  public ResponseEntity<EuropeanaApiErrorResponse> handleCommonHttpException(
+	      HttpException e, HttpServletRequest httpRequest) {
+	    // TODO: harmonize the use of HTTP Exceptions and EuropeanaAPIExceptions
+	    EuropeanaApiErrorResponse response =
+	        new EuropeanaApiErrorResponse.Builder(httpRequest, e, stackTraceEnabled())
+	            .setStatus(e.getStatus().value())
+	            .setError(e.getStatus().getReasonPhrase())
+	            .setMessage(i18nService.getMessage(e.getI18nKey(), e.getI18nParams()))
+	            // code only included in JSON if a value is set in exception
+	            .setCode(e.getI18nKey())
+	            .build();
+	    return ResponseEntity.status(e.getStatus())
+	        .headers(createHttpHeaders(httpRequest))
+	        .body(response);
+	  }
 
+	  @ExceptionHandler(NoHandlerFoundException.class)
+	  public ResponseEntity<EuropeanaApiErrorResponse> handleNoHandlerFoundException(
+	      NoHandlerFoundException e, HttpServletRequest httpRequest) {
+	    EuropeanaApiErrorResponse response =
+	        new EuropeanaApiErrorResponse.Builder(httpRequest, e, stackTraceEnabled())
+	            .setStatus(HttpStatus.NOT_FOUND.value())
+	            .setError(HttpStatus.NOT_FOUND.getReasonPhrase())
+	            .setMessage(e.getMessage())
+	            .build();
+	    return ResponseEntity.status(HttpStatus.NOT_FOUND.value())
+	        .contentType(MediaType.APPLICATION_JSON)
+	        .body(response);
+	  }
+	  
+	  public void throwOriginalError(Exception ex) throws Exception {
+	    if(ex.getCause() instanceof ResourceExhaustedException) {
```

Comment:
the handling of this exception needs to be implemented in the TranslationWebService/GoogleTranslateService